timesheet_validation_status/tests/test_timesheet_validation.py

```diff
@@ -0,0 +1,70 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import pytest
+from odoo.exceptions import ValidationError
+from odoo.tests import common
+
+
+class TestTimesheetValidation(common.SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user_group = cls.env.ref('hr_timesheet.group_hr_timesheet_user')
+        cls.manager_group = cls.env.ref('hr_timesheet.group_timesheet_manager')
+
+        cls.user = cls.env.ref('base.user_demo')
+
+        cls.project = cls.env['project.project'].create({
+            'name': 'My Project',
+        })
+        cls.task = cls.env['project.task'].create({
+            'name': 'My Task',
+            'project_id': cls.project.id,
+        })
+        cls.line_1 = cls.env['account.analytic.line'].create({
+            'name': '/',
+            'project_id': cls.project.id,
+            'task_id': cls.task.id,
+            'user_id': cls.user.id,
+        })
+        cls.line_2 = cls.env['account.analytic.line'].create({
+            'name': '/',
+            'project_id': cls.project.id,
+            'task_id': cls.task.id,
+            'user_id': cls.user.id,
+        })
+
+    def test_manager_can_validate_timesheets(self):
+        self.user.groups_id |= self.manager_group
+        self.line_1.sudo(self.user).validate_timesheet_entries()
+        assert self.line_1.validated_timesheet
+
+    def test_user_can_not_validate_timesheets(self):
+        self.user.groups_id |= self.user_group
+        with pytest.raises(ValidationError):
+            self.line_1.sudo(self.user).validate_timesheet_entries()
+
+    def test_validate_multiple_timesheets(self):
+        (self.line_1 | self.line_2).validate_timesheet_entries()
+        assert self.line_1.validated_timesheet
+        assert self.line_2.validated_timesheet
+
+    def test_timesheet_can_be_validated_multiple_times(self):
+        self.line_1.validate_timesheet_entries()
+        self.line_1.validate_timesheet_entries()
+        assert self.line_1.validated_timesheet
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4667819)